### PR TITLE
Documents moved to new buckets due to split or join might trigger a need

### PR DESCRIPTION
**Pull request by toregge**
for bucket move job to reconsider the target buckets.

@geirst, please review

### CODE REVIEW

File: searchcore/src/tests/proton/documentmetastore/documentmetastore_test.cpp [hunk 1]
```diff
@@ -1460,6 +1498,7 @@ TEST("requireThatBucketInfoIsCorrectAfterJoin")
     EXPECT_EQUAL(bi21.getDocumentCount() +
                  bi23.getDocumentCount(),
                  bi11.getDocumentCount());
+    f.assertNotifyCreateBuckets({ f.bid11 });
 }
 
 TEST("requireThatActiveStateIsPreservedAfterJoin")
```
Comment (geirst) [inline searchcore/src/tests/proton/documentmetastore/documentmetastore_test.cpp]: Consider adding check for join test case as well.

File: searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp [hunk 2]
```diff
@@ -1219,6 +1223,25 @@ TEST_F("explicitly active not ready bucket can be moved to ready even if node is
     EXPECT_EQUAL(f._notReady.bucket(3), f.bucketsModified()[0]);
 }
 
+TEST_F("require that notifyCreateBucket works", ControllerFixture)
```
Comment (geirst) [inline searchcore/src/tests/proton/documentdb/documentbucketmover/documentbucketmover_test.cpp]: Consider changing test name to what is expected when it is called, as specified in the commit message ("... causes bucket to be reconsidered by job")

**Comment by geirst** [comment]
Overall looks good, please have a look at my comments

File: searchcore/src/tests/proton/documentmetastore/documentmetastore_test.cpp (hunk 1 above)
Comment (toregge) [reply searchcore/src/tests/proton/documentmetastore/documentmetastore_test.cpp]: this is the join test case.

Comment (geirst) [reply searchcore/src/tests/proton/documentmetastore/documentmetastore_test.cpp]: Sorry, I misread the diff :)

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**